dj/api/nodes.py

```diff
@@ -329,6 +284,75 @@ def create_node(
     return node  # type: ignore
 
 
+@router.post(
+    "/nodes/{name}/{column}/dimension/{dimension}/",
+    response_model=List[Column],
+)
+def add_dimension_to_node(
+    name: str,
+    column: str,
+    dimension: str,
+    session: Session = Depends(get_session),
+) -> List[Column]:
+    """
+    Add a dimension to a column on a node
+    """
+    node = get_node_by_name(session=session, name=name)
+    dimension_node = get_node_by_name(
+        session=session,
+        name=dimension,
+        node_type=NodeType.DIMENSION,
+    )
+
+    target_column = None
+    for node_column in node.current.columns:
+        if node_column.name == column:
+            target_column = node_column
+            node_column.dimension = dimension_node
+            node_column.dimension_id = dimension_node.id
+    if not target_column:
+        raise DJException(
+            message=f"A column with name `{column}` on node `{name}` does not exist.",
+            http_status_code=404,
+        )
+
+    session.add(node)
+    session.commit()
+    session.refresh(node)
+    return node.current.columns
+
+
+@router.post("/nodes/{name}/table/", response_model=TableMetadata)
+def add_table_to_node(
+    name: str, data: CreateTable, *, session: Session = Depends(get_session)
+) -> TableMetadata:
+    """
+    Add a table to a node
+    """
+    node = get_node_by_name(session=session, name=name)
+    database = get_database_by_name(session=session, name=data.database_name)
+    table = Table(
+        catalog=data.catalog,
+        schema_=data.schema_,
+        table=data.table,
+        database_id=database.id,
+        cost=data.cost,
+        columns=[
+            Column(name=column.name, type=ColumnType(column.type))
+            for column in data.columns
+        ],
+    )
+    session.add(table)
+    session.commit()
+    session.refresh(table)
+
+    node.current.tables.append(table)
+    session.add(node)
+    session.commit()
+    session.refresh(node)
+    return node
```

Comment:
Did you mean to say ?
`return node.current.tables`

Reply:
Good catch, I actually think returning the node at all here isn't the right decision. It makes sense if your creating or updating a node, but in this case the user just wants confirmation that the dimension node has been associated with the column. I'm going to change this to be similar to the availability endpoint and have a response of:
```py
JSONResponse(
        status_code=200,
        content={"message": (
                f"Dimension node {dimension} has been successfully "
                f"linked to column {column} on node {name}"
            )}
    )
```